src/consensus/tx_check.cpp

```diff
@@ -38,10 +38,17 @@ bool CheckTransaction(const CTransaction& tx, TxValidationState& state)
     // of a tx as spent, it does not check if the tx has duplicate inputs.
     // Failure to run this check will result in either a crash or an inflation bug, depending on the implementation of
     // the underlying coins database.
-    std::set<COutPoint> vInOutPoints;
-    for (const auto& txin : tx.vin) {
-        if (!vInOutPoints.insert(txin.prevout).second)
+    if (tx.vin.size() == 2) {
+        if (tx.vin[0].prevout == tx.vin[1].prevout) [[unlikely]] {
```

Comment:
> to allow the compiler to optimize for the case where paths of execution including that statement are less likely than _**any alternative path**_ of execution that does not include such a statement

See here for why this is probably a bad idea: https://blog.aaronballman.com/2020/08/dont-use-the-likely-or-unlikely-attributes/


tl;dr: It's impl-defined whether or not this makes the parent `if(tx.vin.size() == 2)` unlikely as well.

Reply:
Thanks for the context!
Do you think it makes sense to update that here, or should I just close the PR and leave https://github.com/bitcoin/bitcoin/pull/31699 as a follow-up?

Reply:
Fixed, thanks